### PR TITLE
Set up backend / refactor server.js

### LINKED ISSUE #25: Add a MongoDB backend to the repository

### PR DESCRIPTION
**Pull request by jerrylee17**
Resolves #25

### CODE REVIEW

**Review by hsorathia — CHANGES_REQUESTED** [review]
For the comments on the db stuff look at this:
image

Overall, I think we can use more specific types in our database so documents stored into our collection will have more predictable values for their attributes. So even if we have good coding standards and enforce the policies within the code like we make sure we store date objects inside the lastTalked attribute, we can just mess up in one place and accidentally store something else that won't be caught by mongoose. Alternatively, we can use mongoose-timestamps and use the plugin to have an updatedat type field for the model and everytime we update an entity/object mongoose will automatically change the date for us.

Another thing I was thinking about is that the poitns schema seems to be doing a lot (it stores user information and their points). Then it might be better to change the name of this schema to User. Or, we could separate this into two schemas (user and points) and in points we have a reference to the user and vice-versa to establish a relationship between the two. With the latter method, if we decide to add more things like if we have an sce item tracker on the bot (that tracks if the user is currently borrowing an sce item like a logicboard) we can setup a new schema and then reference the user thing to it.

File: server.js [hunk 1]
```diff
@@ -1,19 +1,60 @@
...
+    autoIndex: true,
+    poolSize: 50,
+    bufferMaxEntries: 0,
+    keepAlive: 120,
+    useNewUrlParser: true,
+    useUnifiedTopology: true
+  });
 
-client.login(API_TOKEN);
+  mongoose.set('useCreateIndex', true);
```
[the remark this answers is not in the document]
Comment (jerrylee17) [reply server.js:36]: i dont know how to solve this

File: server.js [hunk 2]
```diff
@@ -1,19 +1,60 @@
...
+  mongoose.connection.once('open', () => {
+    console.log('Connected to Mongo');
```
[the remark this answers is not in the document]
Comment (jerrylee17) [reply server.js:38]: My dumbass used await on the connect lmao

File: api/models/points.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const mongoose = require('mongoose');
+const { composeMongoose } = require('graphql-compose-mongoose');
+
+
+const PointSchema = mongoose.Schema(
+  {
+    // _id: mongoose.Schema.Types.ObjectId,
```
Comment (hsorathia) [inline api/models/points.js:7]: dont need this comment anymore tbh

**Review by hsorathia — APPROVED** [review]

**Pull request merged**